addons/website_slides/models/slide_channel.py

```diff
@@ -660,8 +651,15 @@ def _remove_membership(self, partner_ids):
             ('partner_id', 'in', list(earned_karma)),
         ])
         for user in users:
-            if earned_karma[user.partner_id.id]:
-                user.add_karma(-1 * earned_karma[user.partner_id.id])
+            to_remove = 0
+            channel_id = 0
+            for entry in earned_karma[user.partner_id.id]:
+                to_remove += entry['karma']
+                channel_id = entry['channel_id']
+            entries = earned_karma[user.partner_id.id]
+            to_remove = sum(entry['karma'] for entry in entries)
+            channel_id = entries[-1]['channel_id'] if entries else 0
```

Comment:
Why only the last one ?

Reply:
I think we should loop over the channel, otherwise the origin might be wrong if we call this method in batch